clients/go/main.go

```diff
@@ -58,18 +63,26 @@ func main() {
 
 		log.Printf("Upload offset is: %d", offset)
 
+		uploadCompleteHeader := "?0"
+
+		if(offset == fileInfo.Size()) {
+			log.Println("File is already uploaded")
+			uploadCompleteHeader = "?1"
+		}
+		
+
 		if _, err := file.Seek(offset, os.SEEK_SET); err != nil {
 			log.Fatalf("failed to seek to offset: %s", err)
 		}
 
-		response, err = uploadAppendingProcedure(uploadUrl, file, offset)
+		response, err = uploadAppendingProcedure(uploadUrl, file, offset, uploadCompleteHeader)
 		if err != nil {
 			log.Fatalf("upload creation procedure failed: %s", err)
 		}
 	}
 
 	log.Println("Response from server is:")
-	fmt.Println(response)
+	log.Println(response)
```

Comment:
We didn't use `log.Println` on purpose to now prepend a timestamp and instead write the response as it is to stdout.